driver/driver.go

```diff
@@ -54,17 +44,25 @@ func New(url string) (Driver, error) {
 		return nil, err
 	}
 
-	if d, found := driverMap[u.Scheme]; found {
+	driver := registry.GetDriver(u.Scheme)
+	if driver != nil {
+		blankDriver := reflect.New(reflect.TypeOf(driver)).Interface()
```

Comment:
This is not very idiomatic Go. I think a better solution would be to maintain the old behavior, where drivers register an instance instead of their type, and make the registry a `map[string]Driver`